community_supplied/System/snmp-system-cpu-memory.rule

```diff
@@ -0,0 +1,209 @@
+            }
+            field dt-buffer-memory-utilization {
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;
+                        learning-period 7d;
+                        pattern-periodicity 1d,1h;
+                        field-name "$system-buffer-memory";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for memory buffer utilization";
+            }
+            field dt-cpu-utilization {
+                formula {
+                    dynamic-threshold {
+                        algorithm 3sigma;           
+                        learning-period 7d;
+                        pattern-periodicity 1d,1h;
+                        field-name "$system-cpu";
+                    }
+                }
+                type integer;
+                description "dynamic threshold value for system cpu utilization";
+            }
+            field system-buffer-memory {
+                sensor system-cpu-memory {
+                    path jnxOperatingBuffer;
+                }
+                type integer;
+                description "System memory buffer utilization";
+            }
+            field system-cpu {
+                sensor system-cpu-memory {
+                    path jnxOperatingCPU;
+                }
+                type integer;
+                description "System CPU utilization";
+            }
+            field threshold {
+                constant {
+                    value "{{static-threshold}}";
+                }
+                type integer;
+				description "System CPU utilization"; 
+            }
+			/*
+            * Anomaly detection logic for buffer memory utilization.
+            */
+            trigger system-buffer {
+			    /*
+                * Sets color to red and sends out an anomaly notification when
+                * memory buffer utilization count increases more than threshold static ($threshold) value
+                * 
+                */ 
+                frequency 300s;                      
+                term is-system-buffer-memory-utilization-is-abnormal {
+                    when {
+                        greater-than "$system-buffer-memory" "$threshold";
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "System buffer utilization for $description has crossed static threshold value ($threshold), Value=($system-buffer-memory)";
+                        }
+                    }
+                }
+				/*
+                * Sets color to yellow and sends out an anomaly notification when
+                * memory buffer utilization count increases more than dynamic threshold value
+                * 
+                */
+                term is-system-buffer-memory-utilization-is-minimal {

```

Comment:
Please rephrase with dynamic-threshold check rather than "minimal".

Reply:
changed to "is-buffer-utilization-above-dynamic-threshold"